core/domain/state_domain_test.py

```diff
@@ -5120,3 +5297,1581 @@ def test_validate_negative_duration_seconds(self):
+        self.assertEqual(
+            numeric_input_proto.hints[0].hint_content.text,
+            '<p>This is a copyright character ©.</p>')
+        self.assertEqual(
+            numeric_input_proto.default_outcome.destination_state,
+            'Introduction')
+        self.assertEqual(
+            numeric_input_proto.default_outcome.feedback.content_id,
+            'default_outcome')
+        self.assertEqual(
+            numeric_input_proto.default_outcome.feedback.text,
+            '<p> Default Outcome </p>')
+        self.assertEqual(
+            numeric_input_proto.default_outcome.labelled_as_correct,
+            False)
+        self.assertEqual(
+            numeric_input_proto.answer_groups[0].base_answer_group.outcome.destination_state, # pylint: disable=line-too-long
+            'abc')
+        self.assertEqual(
+            numeric_input_proto.answer_groups[0].base_answer_group.outcome.labelled_as_correct, # pylint: disable=line-too-long
+            True)
+        self.assertEqual(
+            numeric_input_proto.answer_groups[0].base_answer_group.outcome.feedback.content_id, # pylint: disable=line-too-long
+            'feedback_2')
+        self.assertEqual(
+            numeric_input_proto.answer_groups[0].base_answer_group.outcome.feedback.text, # pylint: disable=line-too-long
+            '<p>Feedback</p>')
+        self.assertEqual(
+            numeric_input_proto.answer_groups[0].rule_specs[0].equals.input, # pylint: disable=line-too-long
+            5.0)
+        self.assertEqual(
+            numeric_input_proto.answer_groups[1].rule_specs[0].is_less_than.input, # pylint: disable=line-too-long
+            5.0)
+        self.assertEqual(
+            numeric_input_proto.answer_groups[2].rule_specs[0].is_greater_than.input, # pylint: disable=line-too-long
+            5.0)
+        self.assertEqual(
+            numeric_input_proto.answer_groups[3].rule_specs[0].is_less_than_or_equal_to.input, # pylint: disable=line-too-long
+            5.0)
+        self.assertEqual(
+            numeric_input_proto.answer_groups[4].rule_specs[0].is_greater_than_or_equal_to.input, # pylint: disable=line-too-long
+            5.0)
+        self.assertEqual(
+            numeric_input_proto.answer_groups[5].rule_specs[0].is_inclusively_between.inputLowerInclusive, # pylint: disable=line-too-long
+            5.0)
+        self.assertEqual(
+            numeric_input_proto.answer_groups[5].rule_specs[0].is_inclusively_between.inputUpperInclusive, # pylint: disable=line-too-long
+            10.0)
+        self.assertEqual(
+            numeric_input_proto.answer_groups[6].rule_specs[0].is_within_tolerance.inputTolerance, # pylint: disable=line-too-long
+            5.0)
+        self.assertEqual(
+            numeric_input_proto.answer_groups[6].rule_specs[0].is_within_tolerance.inputComparedValue, # pylint: disable=line-too-long
+            5.0)
+
+
+class RatioExpressionInputInteractionTests(test_utils.GenericTestBase):
```

Comment:
Shouldn't tests for specific interactions go somewhere in extensions/interactions?

Reply:
Shifted to there specific packages and added to shard json file as well.